src/main/java/org/spongepowered/common/data/processor/data/entity/SneakingDataProcessor.java

```diff
@@ -0,0 +1,121 @@
+import org.spongepowered.api.data.manipulator.mutable.entity.SneakingData;
+import org.spongepowered.api.data.merge.MergeFunction;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.ImmutableDataCachingUtil;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeSneakingData;
+import org.spongepowered.common.data.processor.common.AbstractSpongeDataProcessor;
+
+public class SneakingDataProcessor extends AbstractSpongeDataProcessor<SneakingData, ImmutableSneakingData> {
+
+    @Override
+    public Optional<SneakingData> createFrom(DataHolder dataHolder) {
+        if (dataHolder instanceof Entity) {
+            final boolean sneaking = ((Entity) dataHolder).isSneaking();
+            return Optional.<SneakingData>of(new SpongeSneakingData(sneaking));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof Entity;
+    }
+
+    @Override
+    public Optional<SneakingData> from(DataHolder dataHolder) {
+        if (dataHolder instanceof Entity) {
+            final boolean sneaking = ((Entity) dataHolder).isSneaking();
+            return Optional.<SneakingData>of(new SpongeSneakingData(sneaking));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<SneakingData> fill(DataHolder dataHolder, SneakingData manipulator, MergeFunction overlap) {
+        if (dataHolder instanceof Entity) {
+            final Optional<SneakingData> oldData = from(dataHolder);
+            final SneakingData newData = checkNotNull(overlap, "Merge function was null!").merge(oldData.orNull(), manipulator);
+            final Value<Boolean> newValue = newData.sneaking();
+            return Optional.of(manipulator.set(newValue));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<SneakingData> fill(DataContainer container, SneakingData sneakingData) {
+        sneakingData.set(Keys.IS_SNEAKING, getData(container, Keys.IS_SNEAKING));
+        return Optional.of(sneakingData);
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, SneakingData manipulator, MergeFunction function) {
+        if (dataHolder instanceof Entity) {
+            final ImmutableValue<Boolean> newValue = manipulator.sneaking().asImmutable();
+            final SneakingData old = from(dataHolder).get();
+            final ImmutableValue<Boolean> oldValue = old.asImmutable().sneaking();
+            final SneakingData newData = checkNotNull(function, "function").merge(old, manipulator);
+            final boolean sneaking = newData.sneaking().get();
+            try {
+                ((Entity) dataHolder).setSneaking(sneaking);
+                return DataTransactionBuilder.successReplaceResult(newValue, oldValue);
```

Comment:
Please read the javadocs on `DataTransactionBuilder`. It clearly states that the first argument is the "replaced" value and the second argument is the "successful" value. Flip the values.